### PR TITLE
feat!: add PatchDevice and DeviceExistsForName and refactor UpdateDeviceOperatingState

### LINKED ISSUE #1450: Duplicated function to set device operating state
https://github.com/edgexfoundry/device-sdk-go/blob/c2fb876fcd1e0284408ad645a921007a1ace85f5/pkg/service/manageddevices.go#L105-L141

Is there a reason for both? Should one of them be removed? Is one of them deprecated?

### PR DESCRIPTION
**Pull request by ajcasagrande**
BREAKING CHANGE: SetDeviceOperatingState has been removed, and UpdateDeviceOperatingState has been modified to accept a models.OperatingState value.

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [X] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>
https://github.com/edgexfoundry/edgex-docs/pull/1059

## New Dependency Instructions (If applicable)

Closes #1450

### CODE REVIEW

**Review by lenny-intel — DISMISSED** [review]
LGTM, @cloudxxx8 please review.

File: pkg/interfaces/service.go [hunk 1]
```diff
@@ -127,3 +133,10 @@ type DeviceServiceSDK interface {
 	// github.com/rcrowley/go-metrics
 	MetricsManager() interfaces.MetricsManager
 }
+
+// DeviceServiceSDKExt defines the extended contract for device sdk implementations that
+// provide additional APIs needed only from the sdk code.
+type DeviceServiceSDKExt interface {
```
Comment (lenny-intel) [inline pkg/interfaces/service.go:139]: Why is this needed? Why can't Run() be part of the standard API above, Like it is in the App ADK.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — APPROVED** [review]

File: pkg/service/service.go [hunk 2]
```diff
@@ -83,7 +84,7 @@ func NewDeviceService(serviceKey string, serviceVersion string, driver interface
 	service.driver = driver
 
 	service.config = &config.ConfigurationStruct{}
-	return &service, nil
+	return interfaces.DeviceServiceSDKExt(&service), nil
```
Comment (cloudxxx8) [inline pkg/service/service.go:87]: we should move `Run()` to the regular DeviceService interface to make it simpler

Comment (lenny-intel) [reply pkg/service/service.go:87]: @cloudxxx8 , Yes. :-)

Comment (ajcasagrande) [reply pkg/service/service.go:87]: This is fine with me. I modeled it this way based on the SecretProvider from go-mod-bootstrap, as I figured Run() would not be something we expose. If fine exposing it, then yes I can make that change.

File: pkg/interfaces/service.go (hunk 1 above)
Comment (ajcasagrande) [reply pkg/interfaces/service.go]: The app sdk expects the user to call `Run` directly. This sdk expects the user to call `startup.Bootstrap()`.

**Review by lenny-intel — APPROVED** [review]
LGTM
